project3/participant/input_parser/input_parser.cpp

```diff
@@ -3,7 +3,15 @@
 namespace participant::input_parser {
 
 InputParser::InputParser()
-    : multi_id_{0}, is_valid_{true}, req_{REG}, message_{} {}
+    : reg_msg_{pub_sub_messages::Register().New()},
```

Comment:
If you're going to use `New()`, you still need to wrap them in a `unique_ptr` so they get deleted correctly. Otherwise, you can just make them normal class members instead of pointers.